migration/template_generator.py

```diff
@@ -0,0 +1,250 @@
+import logging
+import subprocess
+import json
+from dataclasses import dataclass
+
+import utils
+
+
+@dataclass
+class LPassData:
+    url: str
+    username: str
+    password: str
+    otp_secret: str
+    notes: str
+    title: str
+    vault: str
+
+
+class TemplateGenerator:
+    __AVAILABLE_TEMPLATE_TYPES = ["Login", "Secure Note", "Credit Card", "Bank Account"]
+
+    def __init__(self, lpass_raw_data: LPassData):
+        self.template_type = None
+        self.parsed_notes_data = {}
+        self.lpass_raw_data = lpass_raw_data
+
+        if lpass_raw_data.username and lpass_raw_data.password:
```

Comment:
We may run into an issue using the presence of a username and password as criteria for identifying a login. Conceivably someone might have login items missing one or the other but would still be a valid login item. 

We can't rely on URL necessarily either since there could be login items without one. 

Could we solve this by making login item a default case and use the presence of `http://sn` as the criteria for all other, since `http://sn` is  is guaranteed to be there for all non-logins?

Reply:
Agree. Made it a default case.